core/src/main/kotlin/com/alecstrong/sql/psi/core/psi/mixins/OrderByMixin.kt

```diff
@@ -0,0 +1,21 @@
+package com.alecstrong.sql.psi.core.psi.mixins
+
+import com.alecstrong.sql.psi.core.SqlAnnotationHolder
+import com.alecstrong.sql.psi.core.psi.SqlBindExpr
+import com.alecstrong.sql.psi.core.psi.SqlCompositeElementImpl
+import com.alecstrong.sql.psi.core.psi.SqlOrderingTerm
+import com.intellij.lang.ASTNode
+
+internal abstract class OrderByMixin(
+  node: ASTNode
+) : SqlCompositeElementImpl(node),
+    SqlOrderingTerm {
+  override fun annotate(annotationHolder: SqlAnnotationHolder) {
+    super.annotate(annotationHolder)
+
+    val childBindExpr = node.psi.children.find { it is SqlBindExpr }
+    if (childBindExpr != null) {
+      annotationHolder.createErrorAnnotation(childBindExpr, "Cannot bind the name of a column in an ORDER BY clause")
+    }
```

Comment:
i think you might be able to be more explicit here

```suggestion
    if (expr is SqlBindExpr) {
      annotationHolder.createErrorAnnotation(expr, "Cannot bind the name of a column in an ORDER BY clause")
    }
```